ts/src/binance.ts

```diff
@@ -8598,31 +8600,32 @@ export default class binance extends Exchange {
             // despite that their message is very confusing, it is raised by Binance
             // on a temporary ban, the API key is valid, but disabled for a while
             if ((error === '-2015') && this.options['hasAlreadyAuthenticatedSuccessfully']) {
-                throw new DDoSProtection (this.id + ' ' + body);
+                throw new DDoSProtection (errorMessage);
             }
-            const feedback = this.id + ' ' + body;
             if (message === 'No need to change margin type.') {
                 // not an error
                 // https://github.com/ccxt/ccxt/issues/11268
                 // https://github.com/ccxt/ccxt/pull/11624
                 // POST https://fapi.binance.com/fapi/v1/marginType 400 Bad Request
                 // binanceusdm {"code":-4046,"msg":"No need to change margin type."}
-                throw new MarginModeAlreadySet (feedback);
+                throw new MarginModeAlreadySet (errorMessage);
             }
-            this.throwExactlyMatchedException (this.exceptions['exact'], error, feedback);
-            throw new ExchangeError (feedback);
+            this.throwExactlyMatchedException (this.exceptions['exact'], error, errorMessage);
+            throw new ExchangeError (errorMessage);
         }
+        // If we are here, error and eventually code are undefined
         if (!success) {
-            throw new ExchangeError (this.id + ' ' + body);
+            throw new ExchangeError (message);
         }
         if (Array.isArray (response)) {
             // cancelOrders returns an array like this: [{"code":-2011,"msg":"Unknown order sent."}]
             const arrayLength = response.length;
             if (arrayLength === 1) { // when there's a single error we can throw, otherwise we have a partial success
                 const element = response[0];
                 const errorCode = this.safeString (element, 'code');
+                const errorMessage = this.safeString (element, 'msg');
                 if (errorCode !== undefined) {
-                    this.throwExactlyMatchedException (this.exceptions['exact'], errorCode, this.id + ' ' + body);
+                    this.throwExactlyMatchedException (this.exceptions['exact'], errorCode, errorCode + ': ' + errorMessage);
```

Comment:
`errorMessage` already contains the code, right?

Reply:
No, `errorMessage` was initialized two lines earlier 
`const errorMessage = this.safeString (element, 'msg');`